external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaStreamingWrite.scala

```diff
@@ -103,20 +103,21 @@ class KafkaStreamDataWriter(
     // Send is asynchronous, but we can't commit until all rows are actually in Kafka.
     // This requires flushing and then checking that no callbacks produced errors.
     // We also check for errors before to fail as soon as possible - the check is cheap.
-    checkForErrors()
-    producer.flush()
-    checkForErrors()
+    close()
```

Comment:
@zsxwing and @gaborgsomogyi, Doing a flush during a commit can have an impact on the overall performance of the KafkaSink. I have not changed it, from what it was, but it is worth mentioning.

Reply:
If my understanding is correct, this is to prevent data loss where Spark marks the offset as committed but writing to Kafka finally failed asynchronously: I think performance doesn't matter here.